article/helpers.py

```diff
@@ -7,128 +7,81 @@
         self.request = request
-        self.current_article = current_article
-
-    @abc.abstractmethod
-    def retrieve_historic_article_uuids(self):
-        return
 
-    def get_group_read_progress(self):
+    def view_progress_for_groups(self):
         return {
             group.name: {
-                'read': len(self.read_article_uuids & group.articles_set),
-                'total': len(group.articles_set),
+                'read': len(self.viewed_article_uuids & group.article_uuids),
+                'total': len(group.article_uuids),
             }
             for group in structure.ALL_GROUPS
         }
 
-    def read_articles_keys_in_group(self, group_name):
-        article_uuids = structure.get_article_group(group_name).articles_set
-        # read_articles_in_category is a new set (intersection)
-        # with elements common to read_articles and articles_in_category
-        read_articles_in_group = self.read_article_uuids & article_uuids
-        return read_articles_in_group
-
-    def article_read_count(self, group_name):
-        return len(self.read_articles_keys_in_group(group_name))
+    def articles_viewed_for_group(self, group_name):
+        group = structure.get_article_group(group_name)
+        return self.viewed_article_uuids & group.article_uuids
 
-    def remaining_reading_time_in_group(self, group_name):
-        """ Return the remaining reading time in minutes
-            for unread articles in the group
-        """
-
-        read_articles_uuids_in_group = self.read_articles_keys_in_group(
-            group_name
-        )
-        read_articles = structure.get_articles_from_uuids(
-            read_articles_uuids_in_group
-        )
-        read_articles_total_time = total_time_to_read_multiple_articles(
-            read_articles
-        )
-        group_total_reading_time = structure.get_article_group(
-            group_name
-        ).total_reading_time
-        return round(group_total_reading_time - read_articles_total_time, 2)
+    def remaining_view_time_for_group(self, group_name):
+        group = structure.get_article_group(group_name)
+        return group.remaining_time_to_view(self.viewed_article_uuids)
 
     @cached_property
-    def read_article_uuids(self):
-        uuids = self.retrieve_historic_article_uuids()
-        if self.current_article:
-            uuids.add(self.current_article.uuid)
-        return uuids
+    def viewed_article_uuids(self):
+        return self.retrieve_viewed_article_uuids()
+
+    @abc.abstractmethod
+    def retrieve_viewed_article_uuids(self):
+        return
 
 
-class ArticleReadManager:
+class ArticlesViewedManager:
     def __new__(cls, request, current_article=None):
-        session_manager = SessionArticlesReadManager(request, current_article)
-        database_manager = DatabaseArticlesReadManager(
-            request, current_article
-        )
+        session_manager = SessionArticlesViewedManager(request)
```

Comment:
it is a bit of unexpected that instantiating a class can cause DB reads and writes. But I understand this is to handle the complexity of syncing session and db progress.

To keep it explicit let's split the code below into 2 paths - user logged in and not (e.g.

```
if request.sso_user is None:
    articles_viewed_manager = self.get_session_articles_viewed_manager(request, current_article)
else:
    articles_viewed_manager = self.get_database_articles_viewed_manager(request, current_article)

return articles_viewed_manager
```